prompt2model/param_selector/search_with_optuna.py

```diff
@@ -0,0 +1,245 @@
+"""This module provides automatic hyperparameter selection using optuna."""
+
+from __future__ import annotations  # noqa FI58
+
+import os
+from typing import Any, Optional
+
+import optuna
+import transformers
+from datasets import Dataset, concatenate_datasets
+from optuna.trial import Trial
+from transformers import PreTrainedModel  # noqa
+from transformers import PreTrainedTokenizer
+
+from prompt2model.model_trainer.generate import GenerationModelTrainer
+from prompt2model.param_selector.base import ParamSelector
+from prompt2model.prompt_parser.base import PromptSpec
+from prompt2model.utils.config import DEFAULT_HYPERPARAMETERS_SPACE
+
+
+class OptunaParamSelector(ParamSelector):
+    """Uses optuna for searching for hyperparameters."""
+
+    def __init__(self, trainer: GenerationModelTrainer, n_trials: int):
+        """Initialize with train/val datasets and a prompt specification.
```

Comment:
I think this docstring is incorrect? There are no datasets or prompt spec provided - please update

Reply:
Let me know if this wordings sounds right.

Reply:
Looks good